src/java/com/google/tagpost/spanner/DataService.java

```diff
@@ -10,5 +8,9 @@
 
   List<Thread> getAllThreadsByTag(String tag);
 
-  List<Comment> getAllCommentsByThreadId(long threadId);
+  Thread addNewThreadWithTag(Thread thread) throws Exception;
```

Comment:
How about we do not declare throwing exception here in `DataService.java`, to avoid making the interface too restrictive.

In the actual implementation `SpannerService.java`  , `addNewThreadWithTag(Thread thread)` thorws an `SpannerException`

Reply:
If I understand correctly, that cannot pass java static exception check if the inherited function throws exception. One solution is to throw a new class of RuntimeException instead of Exception (but do not throw RuntimeException itself since it is an anti-pattern), so you do not need to throw checked exception.